conanfile.py

```diff
@@ -0,0 +1,72 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd, can_run
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.scm import Git
+
+class cpprealmRecipe(ConanFile):
+    name = "cpprealm"
+    version = "1.0"
+
+    # Optional metadata
+    license = "Apache-2.0"
+    url = "https://github.com/realm/realm-cpp"
+    description = "Realm is a mobile database that runs directly inside phones, tablets or wearables."
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False]}
+    default_options = {"shared": False}
+
+    def is_darwin(self):
+        return self.settings.os == "Macos" or self.settings.os == "iOS" or self.settings.os == "watchOS"
+
+    def validate(self):
+        check_min_cppstd(self, "17")
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+        if not self.is_darwin() and not self.settings.os == "Emscripten":
+            self.requires("zlib/1.3")
+        if not self.is_darwin():
+            self.requires("openssl/3.2.0")
+            self.requires("libuv/1.48.0")
+        if self.settings.os == "Linux":
+            self.requires("libcurl/8.4.0")
+    def source(self):
+        git = Git(self)
+        git.clone(url="https://github.com/realm/realm-cpp", target=".")
+        git.folder = "."
+        git.checkout(commit="316c31841b56fab7fe1f28f0aa0138e00f10a8cb")
```

Comment:
This temporarily points to the changelog commit in this PR. This should point to a v2.0.0 tag in the future